packages/php-wasm/universal/src/lib/php-request-handler.ts

```diff
@@ -306,14 +319,89 @@ export class PHPRequestHandler {
 			),
 			this.rewriteRules
 		);
-		const fsPath = joinPaths(this.#DOCROOT, normalizedRequestedPath);
-		if (!seemsLikeAPHPRequestHandlerPath(fsPath)) {
-			return this.#serveStaticFile(
-				await this.processManager.getPrimaryPhp(),
-				fsPath
-			);
+
+		const primaryPhp = await this.getPrimaryPhp();
+
+		let fsPath = joinPaths(this.#DOCROOT, normalizedRequestedPath);
+
+		if (primaryPhp.isDir(fsPath)) {
+			// Ensure directory URIs have a trailing slash. Otherwise,
+			// relative URIs in index.php or index.html files are relative
+			// to the next directory up.
+			//
+			// Example:
+			// For a request to "/wp-admin", the relative link "edit.php"
```

Comment:
The logic seems to be generic enough to belong to PHPRequestHandler, but the comments talk all about WordPress. Let's document this behavior in a generic, non-WordPress way and then add a note at the bottom "by the way, here's three sentences why WordPress needs that, too.

Reply:
I've updated the notes to be more generic and ended with a short note about how it matters for WordPress as well.